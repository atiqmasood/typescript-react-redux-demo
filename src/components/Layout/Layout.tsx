import * as React from "react";
import { Grid, Menu } from "semantic-ui-react";

export interface IProps {
  children?: React.ReactNode;
}
/**
 * Application layout component
 */
export default class Layout extends React.Component<IProps> {
  constructor(props: IProps) {
    super(props);

    this.state = {};
  }

  public render() {
    return (
      <Grid>
        <Grid.Row columns={1}>
          <Grid.Column>
            <Menu>
              <Menu.Item header={true}>Online Store</Menu.Item>
              <Menu.Item name="demo" />
            </Menu>
          </Grid.Column>
        </Grid.Row>

        <Grid.Row columns={1}>
          <Grid.Column>
            <div className="body">{this.props.children}</div>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}
