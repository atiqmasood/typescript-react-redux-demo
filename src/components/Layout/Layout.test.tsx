import * as React from "react";
import Layout from "./Layout";

import * as renderer from "react-test-renderer";

describe("Layout component renders correctly", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(<Layout />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
