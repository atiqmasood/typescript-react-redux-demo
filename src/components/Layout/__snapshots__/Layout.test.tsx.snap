// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly renders correctly 1`] = `
<div
  className="ui grid"
>
  <div
    className="one column row"
  >
    <div
      className="column"
    >
      <div
        className="ui menu"
      >
        <div
          className="header item"
          onClick={[Function]}
        >
          AboutYou
        </div>
        <div
          className="item"
          onClick={[Function]}
        >
          Demo
        </div>
      </div>
    </div>
  </div>
  <div
    className="one column row"
  >
    <div
      className="column"
    >
      <div
        className="body"
      />
    </div>
  </div>
</div>
`;
